lib/modules/api/response.rb

```diff
@@ -282,6 +282,12 @@ def add_paging_and_sorting(new_query, filter_settings, filter_query)
         # execute a count against entire set without paging
         total = new_query.size
 
+        # if query involves aggregation, size returns the size of each group.
+        # and what we need is the number of groups
+        if total.is_a? Hash
```

Comment:
this case seems to be exclusive with the `.size` query above. as such the should be in an `elseif`